File: C Language/Merge_Two_Sorted_LinkedList.c
//Remove Duplicates from Sorted Singly Linked List

#include<stdio.h>
#include<stdlib.h>

struct node
{
    int data;
    struct node * next;
    struct node * prev;
};

struct node *head1 = NULL;
struct node *head2 = NULL;

void linkedListTraversal(struct node* head)
{
    while(head != NULL)
    {
        printf("%d ", head -> data);
        head = head -> next;
    }
}

void createList(struct node** head, int data)
{
    struct node *ptr = (struct node*)malloc(sizeof(struct node));
    ptr -> data = data;
    ptr -> next = *head;	
    (*head) = ptr;
}

struct node * MergeLinkedList(struct node* h1, struct node* h2)
{
    struct node *dummy = (struct node*)malloc(sizeof(struct node));
    struct node *tail = dummy;
    while(1)
    {
        if(h1 == NULL)
        {
            tail -> next = h2;
            break;
        }
        else if(h2 == NULL)
        {
            tail -> next = h1;
            break;
        }
        if(h1 -> data <= h2 -> data)
        {
				        tail -> next = h1;
				        h1 = h1 -> next;
        }
				    else
        {
				        tail -> next = h2;
				        h2 = h2 -> next;
        }
				tail = tail -> next;
    }
    return dummy -> next;
}


int main()
{
   
    createList(&head1, 33);
    createList(&head1, 27);
    createList(&head1, 20);
    createList(&head1, 14);
    createList(&head1, 8);
    createList(&head1, 3);
    printf("Linked List 1 is\n");
    linkedListTraversal(head1);
    
    printf("\n\n");
    
    createList(&head2, 37);
    createList(&head2, 29);
    createList(&head2, 22);
    createList(&head2, 18);
    createList(&head2, 10);
    createList(&head2, 5);
    
    printf("Linked List 2 is\n");
    linkedListTraversal(head2);
    struct node * head = MergeLinkedList(head1, head2);
    printf("\n\nMerged Linked List is\n");
    linkedListTraversal(head);
    return 0;
}

File: Python/Merge_Two_Sorted_LinkedList.py
class Node:
	def __init__(self, data):
		self.data = data
		self.next = None
		self.prev = None

class LinkedList:
	def __init__(self):
		self.head = None
			
	def push(self, data):
		newNode = Node(data)
		newNode.next = self.head
		self.head = newNode
		
	def printList(self, ptr):
		ptr = self.head
		if ptr is None:
			print('List has no element')
			return
		else:
			temp = ptr
			while(temp):
				print(temp.data, end=' ')
				temp = temp.next
				
def MergeLinkedList(head1, head2):
		dummy = Node(None)
		tail = dummy
		while True:
				if head1 is None:
					tail.next = head2
					break
				elif head2 is None:
					tail.next = head1
					break
				if head1.data <= head2.data:
				        tail.next = head1
				        head1 = head1.next
				else:
				        tail.next = head2
				        head2 = head2.next
				tail = tail.next
		return dummy.next
				
LL1 = LinkedList()
LL1.push(33)
LL1.push(24)
LL1.push(20)
LL1.push(12)
LL1.push(6)
LL1.push(4)
LL1.push(2)

print()

LL2 = LinkedList()
LL2.push(34)
LL2.push(27)
LL2.push(22)
LL2.push(15)
LL2.push(8)
LL2.push(5)
LL2.push(3)

print("Linked List 1 is:")
LL1.printList(LL1.head)

print("\n\nLinked List 2 is:")
LL2.printList(LL2.head)

LL3 = LinkedList()
print('\n')

print("Merged Linked List is: ")
LL3.head = MergeLinkedList(LL1.head, LL2.head)
LL3.printList(LL3.head)